### PR TITLE
i18n(fr): update `api-reference.mdx`

### PR DESCRIPTION
**Pull request by ArmandPhilippot**
#### Description (required)

* Apply changes from #8943 to French translation of `api-reference.mdx`
* Reword a few sentences to make reading easier.
* Apply fix (missing brackets) from #9006

#### Related issues & labels (optional)

- Suggested label: i18n

### CODE REVIEW

File: src/content/docs/fr/reference/api-reference.mdx [hunk 1]
```diff
@@ -1347,6 +1536,29 @@ const enterprise = await getEntryBySlug('blog', 'enterprise');
...
 
+### `getDataEntryById()`
+
+<p>
+
+**Type :** `(collection: string, id: string) => Promise<CollectionEntry<collection>>`<br />
+<Since v="2.5.0" />
+</p>
+
+:::caution[Deprecated]
```
Comment (thomasbnt) [inline src/content/docs/fr/reference/api-reference.mdx:1547]:
```suggestion
:::caution[Obsolète]
```

**Review by thomasbnt — CHANGES_REQUESTED** [review]
One word and LGTM!

**Review by thomasbnt — APPROVED** [review]
LGTM !

File: src/content/docs/fr/reference/api-reference.mdx [hunk 2]
```diff
@@ -1642,7 +1903,10 @@ getPathByLocale("fr-CA"); // renvoie "french"
 
 ### `getLocaleByPath`
```
Comment (thomasbnt) [inline src/content/docs/fr/reference/api-reference.mdx:1904]:
```suggestion
### `getLocaleByPath()`
```

Edit from https://github.com/withastro/docs/pull/9006

**Review by thomasbnt — CHANGES_REQUESTED** [review]

**Review by thomasbnt — APPROVED** [review]

**Review by yanthomasdev — APPROVED** [review]
LGTM! Good work!

**Pull request merged**